geoapps/io/PeakFinder/params.py

```diff
@@ -339,15 +339,16 @@ def update(self, params_dict: Dict[str, Any]):
                 for param in self._free_param_keys:
                     if param in key.lower():
                         group = key.lower().replace(param, "").rstrip()
+                        group = group.replace(self._free_param_identifier, "").lstrip()
 
                         if group not in free_param_dict:
                             free_param_dict[group] = {}
 
                         free_param_dict[group][param] = value
-                        self.default_ui_json[key] = self.default_ui_json[
+                        self.default_ui_json[key] = default_ui_json[
```

Comment:
I recently went through the code and protected the original import of default_ui_json with a deepcopy everywhere.  This line seems like it might leave us vulnerable to having references in the original changed at runtime - tricky to debug!

Maybe throw another deepcopy in there to be safe?
```suggestion
                        self.default_ui_json[key] = deepcopy(default_ui_json[f"Template {param.capitalize()}"])
```